it/src/main/java/com/google/cloud/teleport/it/spanner/SpannerResourceManagerUtils.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.spanner;
+
+import static com.google.common.hash.Hashing.goodFastHash;
+import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.base.CharMatcher;
+import com.google.common.hash.HashFunction;
+import com.google.re2j.Pattern;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Random;
+
+/**
+ * Utilities for {@link com.google.cloud.teleport.it.spanner.SpannerResourceManager}
+ * implementations.
+ */
+public final class SpannerResourceManagerUtils {
+  private static final DateTimeFormatter TIME_FORMAT =
+      DateTimeFormatter.ofPattern("yyyyMMdd-HHmmss-SSSSSS");
+  private static final Pattern ILLEGAL_DATABASE_CHARS = Pattern.compile("[\\W-]");
```

Comment:
Can I do this since it takes care of all illegal characters at once or should I leave it as it was and account for only '$', '-' and '.'?

Reply:
You can leave it as it is. I think this will cover most, if not all, illegal characters test authors may actually use. I don't think the other characters are necessary. Most likely, someone chooses a delimiter between words that isn't allowed. I think '$' and '.' could be signs of other problems with the test.